packages/react/src/isomorphicClerk.ts

```diff
@@ -542,12 +543,17 @@ export default class IsomorphicClerk {
     }
   };
 
-  addListener = (listener: (emission: Resources) => void): void => {
-    const callback = () => this.clerkjs?.addListener(listener);
+  addListener = (listener: ListenerCallback): UnsubscribeCallback => {
+    // This is used to resolve the type issue of addListener return value
+    // since callback return value could be undefined due to this.clerkjs?.
+    const noopCallback = () => undefined;
+    const callback = () => this.clerkjs?.addListener(listener) || noopCallback;
+
     if (this.clerkjs) {
-      callback();
+      return callback();
```

Comment:
We do know that the TS warning is purely on the type-level, we can probably just add an assertion here instead of the `noopCallback`
```suggestion
      return callback() as UnsubscribeCallback;
```